utils feature fails to compile without model feature

I'm getting the following error when compiling serenity with the `utils` feature but without the `model` feature:

```
error: cannot find macro `__impl_bitflags` in this scope
   --> /Users/fenhl/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.7.4/src/model/channel/message.rs:856:1
    |
856 | __impl_bitflags! {
    | ^^^^^^^^^^^^^^^

error[E0404]: expected trait, found derive macro `Serialize`
   --> /Users/fenhl/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.7.4/src/model/channel/message.rs:877:6
    |
877 | impl Serialize for MessageFlags {
    |      ^^^^^^^^^
    |
help: a trait with a similar name exists
    |
877 | impl Deserialize for MessageFlags {
    |      ^^^^^^^^^^^
help: possible better candidates are found in other modules, you can import them into scope
    |
5   | use serde::Serialize;
    |
5   | use serde::ser::Serialize;
    |

error[E0405]: cannot find trait `Serializer` in this scope
   --> /Users/fenhl/.cargo/registry/src/github.com-1ecc6299db9ec823/serenity-0.7.4/src/model/channel/message.rs:879:14
    |
879 |     where S: Serializer
    |              ^^^^^^^^^^
    |
help: a trait with a similar name exists
    |
879 |     where S: Deserializer
    |              ^^^^^^^^^^^^
help: possible candidates are found in other modules, you can import them into scope
    |
5   | use serde::Serializer;
    |
5   | use serde::ser::Serializer;
    |
```

# Workaround

Adding the `model` and `rustls_backend` features fixes this, but adds many unnecessary dependencies.

--- a/src/model/channel/message.rs
+++ b/src/model/channel/message.rs
@@ -849,10 +849,12 @@ pub struct ChannelMention {
 
 /// Describes extra features of the message.
 #[derive(Copy, PartialEq, Eq, Clone, PartialOrd, Ord, Hash)]
+#[cfg_attr(not(feature = "model"), derive(Debug, Deserialize, Serialize))]
 pub struct MessageFlags {
     pub bits: u64,
 }
 
+#[cfg(feature = "model")]
 __impl_bitflags! {
     MessageFlags: u64 {
         /// This message has been published to subscribed channels (via Channel Following).
@@ -864,6 +866,7 @@ __impl_bitflags! {
     }
 }
 
+#[cfg(feature = "model")]
 impl<'de> Deserialize<'de> for MessageFlags {
     fn deserialize<D>(deserializer: D) -> StdResult<Self, D::Error>
     where D: Deserializer<'de>
@@ -874,6 +877,7 @@ impl<'de> Deserialize<'de> for MessageFlags {
     }
 }
 
+#[cfg(feature = "model")]
 impl Serialize for MessageFlags {
     fn serialize<S>(&self, serializer: S) -> StdResult<S::Ok, S::Error>
     where S: Serializer

--- a/src/model/user.rs
+++ b/src/model/user.rs
@@ -441,6 +441,7 @@ pub struct User {
 }
 
 use std::hash::{Hash, Hasher};
+#[cfg(feature = "model")]
 use chrono::{DateTime, FixedOffset};
 
 impl PartialEq for User {

